Modules/System/Geolocation/src/GeolocationImpl.Codeunit.al

```diff
@@ -0,0 +1,33 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+codeunit 50101 "Geolocation Impl."
+{
+    Access = Internal;
+
+    var
+        GeolocationPage: Page Geolocation;
+
+    procedure GetGeolocation(var Latitude: Decimal; var Longitude: Decimal): Boolean
+    begin
+        if not IsAvailable() then
+            exit(false);
+
+        Clear(GeolocationPage);
+        GeolocationPage.SetHighAccuracy(true);
+        GeolocationPage.RunModal();
```

Comment:
This is what I do not understand....

So there is a codeunit that can retrieve the location and a page that.... can do the same?

Why have two interfaces for the same thing?

I'd suggest keeping only the codeunit, so the usage would be something of the sort:
```
Geolocation.SetTimeout(3000);
Geolocation.GetHighAccuracy(false);
if Geolocation.GetGeolocation(latitude, longitude) then
    ....
```

Reply:
I still don't see why you need to have both.

Whether finetuned or not, having two ways to use the module's functionality is confusing.

While you cannot entirely hide the page (pages cannot be made internal), as long as you don't expose functionality through it, it's okay.

So, discuss it again with @darjoo and either hide the page's functionality and let the codeunit control it, or remove the facade codeunit and let the page the only available interface.

Reply:
Yes it was indeed confusing. I have moved the functionality away from the page and the module's functionality can now only be accessed through the public facade codeunit.

Example:
```
Geolocation.SetHighAccuracy(true);
if Geolocation.RequestGeolocation() then
    Geolocation.GetGeolocation(Latitude, Longitude);    
```